docs/source/newsfragments/4212.change.rst

```diff
@@ -0,0 +1 @@
+The GPI `get_range` method now returns a range's direction in addition to the left and right bounds. When using VHPI or ModelSim's FLI, the direction is set explicitly by the `to`/`downto` keywords in the range definition; otherwise, the direction is inferred by the relative values of the left and right bounds. This change also allows null range sizes to be inferred correctly in VHDL.
```

Comment:
```suggestion
The GPI ``get_range`` method now returns a range's direction in addition to the left and right bounds. When using VHPI or ModelSim's FLI, the direction is set explicitly by the ``to``/``downto`` keywords in the range definition; otherwise, the direction is inferred by the relative values of the left and right bounds. This change also allows null range sizes to be inferred correctly in VHDL.
```
Maybe also replace "VHPI or ModelSim's FLI" with just "VHDL", we don't need to be too technical here.